File: Topic Based Problems/01 Fundamentals/Beginner/12 Power Calculation/C++/PowerCalculation.cpp
/**
 * @file PowerCalculation.cpp
 * @author SharifdotG (sharifmdyousuf007@gmail.com)
 * @brief Power Calculation in C++
 * @version 1.0
 * @date 2023-09-24
 * 
 * @copyright Copyright (c) 2023
 * 
 */

#include <iostream>
#include <cmath>
#include <iomanip>

using namespace std;

int main() {
    double base, exponent;

    cout << "Enter base: ";
    cin >> base;

    cout << "Enter exponent: ";
    cin >> exponent;

    cout << "Power of " << base << " to the " << exponent << " is: " << fixed << setprecision(2) << pow(base, exponent) << endl;

    return 0;
}

File: Topic Based Problems/01 Fundamentals/Beginner/12 Power Calculation/Java/PowerCalculation.java
/**
 * @file PowerCalculation.java
 * @brief Power Calculation in Java
 * @version 1.0
 * @date 2023-09-24
 * 
 * This program calculates the power of a base to an exponent in Java.
 * 
 * @author SharifdotG (sharifmdyousuf007@gmail.com)
 */

import java.util.Scanner;

public class PowerCalculation {
    public static void main(String[] args) {
        double base, exponent;

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter base: ");
        base = scanner.nextDouble();

        System.out.print("Enter exponent: ");
        exponent = scanner.nextDouble();

        System.out.printf("Power of %.2f to the %.2f is: %.2f\n", base, exponent, Math.pow(base, exponent));

        scanner.close();
    }
}
